### DISCUSSION TITLE
Donation Form UI with intercation

### DISCUSSION THREAD

**Discussion by ST10082857** [IIE-Cape-Tech-Creatives/Phumlani_Dube_Foundation-mobileApp#11] [Show and tell]
### Steps for User Interaction on the Donation Page:

1. **Entering Details:**
   - The user taps on the **name**, **email**, and **donation amount** fields to input their information.
   - The keyboard appears for the user to type in their details.
![Donation Form](https://github.com/user-attachments/assets/7ecb4c73-2c4f-4ab8-b531-6055ff267e1d)

2. **Submitting the Donation:**
   - After entering all information, the user presses the **Submit** button.
   - The app **validates** the input (checks that the name, email, and donation amount are correctly filled).
![Donation Form with Empty Error](https://github.com/user-attachments/assets/77148f81-0e74-4ade-b845-aea813ec1b19)

3. **Processing the Donation:**
   - If all details are valid, the app processes the donation and sends two **confirmation emails**:
     - One email is sent to the business.
     - Another confirmation email is sent to the user who made the donation.

4. **Handling Errors:**
   - If any field is incomplete or incorrect, the app displays an **error message**, asking the user to correct the input.
![Error Handeling](https://github.com/user-attachments/assets/e021502a-7761-4e78-8d52-574813d46e26)

5. **Navigating Back:**
   - The user can press the **Back** button to return to the previous screen without submitting the form.

6. **Feedback:**
   - The app provides feedback throughout, showing confirmation or error messages based on the user’s actions.

**Comment by ST10036066** [comment]
Awesome work Mushfeeq!
